### PR TITLE
Add an identity proof viewer

### PR DESCRIPTION
**Pull request by ianopolous**
This also meant fixing public links to files, and making sure they propagate open=true to the resulting secretlink

### CODE REVIEW

File: src/components/identity-proof-viewer.vue [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                return;
+            if (this.file.isDirectory())
+                return;
+            var props = this.file.getFileProperties();
+            var that = this;
+            this.showSpinner = true;
+            this.file.getInputStream(this.context.network, this.context.crypto, 
+                props.sizeHigh(), props.sizeLow(), 
+                function(read) {})
+                .thenCompose(function(startReader) {
```
Comment (kevodwyer) [inline src/components/identity-proof-viewer.vue:51]: This seems overly complicated. Shouldn't it just read the entire file?

File: src/components/filesystem/index.js [hunk 2]
```diff
@@ -514,6 +523,8 @@ module.exports = {
 		this.showCodeEditor = true;
 	    else if (app == "hex")
 		this.showHexViewer = true;
+	    else if (app == "identity-proof")
+		this.showIdentityProof = true;
```
Comment (kevodwyer) [inline src/components/filesystem/index.js:527]: what file icon is used?

**Review by kevodwyer — COMMENTED** [review]
could this be added to edit/view profile page?

File: src/components/filesystem/index.js (hunk 2 above)
Comment (ianopolous) [reply src/components/filesystem/index.js:527]: the default is fine for now

File: src/components/identity-proof-viewer.vue (hunk 1 above)
Comment (ianopolous) [reply src/components/identity-proof-viewer.vue]: I've simplified it now

**Review by kevodwyer — APPROVED** [review]

**Pull request merged**